core/transactions/staged_mutation.cxx

```diff
@@ -25,6 +25,23 @@
 namespace couchbase::core::transactions
 {
 
+void
+unstaging_state::wait_until_unstage_possible()
+{
+    std::unique_lock lock(mutex_);
```

Comment:
Is mutex_ reentrant?  Because we've already locked it at the top of staged_mutation_queue::commit/rollback.  Just checking that this isn't an accidental no-op.

Reply:
That's the `mutex_` of `unstaging_state` which is different from `mutex_` of the `staged_mutation_queue` that's locked at commit/rollback